skill-tree/src/main/java/com/RDS/skilltree/Authentication/UserAuthenticationToken.java

```diff
@@ -0,0 +1,41 @@
+package com.RDS.skilltree.Authentication;
+
+import com.RDS.skilltree.User.UserModel;
+import com.RDS.skilltree.User.UserRole;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import javax.security.auth.Subject;
+import java.util.List;
+import java.util.UUID;
+
+public class UserAuthenticationToken extends AbstractAuthenticationToken {
+private final String rdsUserId;
+private final UserRole role ;
+
+    public UserAuthenticationToken(String role, String rdsUserId) {
+        super(List.of(new SimpleGrantedAuthority(role)));
+
+        this.rdsUserId = rdsUserId;
+        this.role  = UserRole.fromString(role);
```

Comment:
What happens in the case there is a new role present in the token, and the enum is not updated in code?

Reply:
than this function return null, that means we didn't get the role, and we want the authorisation to stop them

Reply:
But that wouldn't be the behaviour right? The `this.role` would be set to `null`.

Reply:
added a guest role, so if the above scenarios we assign guest role with read-only access